test/unit/repositories/test_system.py

```diff
@@ -24,779 +25,789 @@
+    def test_get_with_invalid_id(self):
+        """Test getting a System with an invalid ID"""
 
-    with pytest.raises(InvalidObjectIdError) as exc:
-        system_repository.update(system_id, MagicMock())
-    assert str(exc.value) == f"Invalid ObjectId value '{system_id}'"
+        system_id = "invalid-id"
 
-    database_mock.systems.update_one.assert_not_called()
-    database_mock.systems.find_one.assert_not_called()
+        self.call_get_expecting_error(system_id, InvalidObjectIdError)
+        self.check_get_failed_with_exception("Invalid ObjectId value 'invalid-id'")
 
 
-def test_update_with_non_existent_parent_id(test_helpers, database_mock, system_repository):
-    """
-    Test updating a System with a non-existent parent ID
+class GetBreadcrumbsDSL(SystemRepoDSL):
+    """Base class for breadcrumbs tests"""
 
-    Verify that the `update` method properly handles the update of a System with a non-existent parent ID
-    """
-    system = SystemIn(**{**SYSTEM_A_INFO, "parent_id": str(ObjectId())})
-    system_id = str(ObjectId())
+    _breadcrumbs_query_result: dict
+    _mock_aggregation_pipeline = MagicMock()
+    _expected_breadcrumbs: MagicMock
+    _obtained_system_id: str
+    _obtained_breadcrumbs: MagicMock
 
-    # Mock `find_one` to not return a parent System document
-    test_helpers.mock_find_one(database_mock.systems, None)
+    def mock_breadcrumbs(self, breadcrumbs_query_result: dict):
+        """Mocks database methods appropriately to test the 'get_breadcrumbs' repo method
 
-    with pytest.raises(MissingRecordError) as exc:
-        system_repository.update(system_id, system)
-    assert str(exc.value) == f"No parent System found with ID: {system.parent_id}"
+        :param breadcrumbs_query_result: Dictionary containing the breadcrumbs query result from the aggregation
+                                         pipeline
+        """
+        self._breadcrumbs_query_result = breadcrumbs_query_result
+        self._mock_aggregation_pipeline = MagicMock()
+        self._expected_breadcrumbs = MagicMock()
 
-    database_mock.systems.update_one.assert_not_called()
-    database_mock.systems.find_one.assert_called_once_with({"_id": system.parent_id}, session=None)
+        self.mock_utils.create_breadcrumbs_aggregation_pipeline.return_value = self._mock_aggregation_pipeline
+        self.systems_collection.aggregate.return_value = breadcrumbs_query_result
+        self.mock_utils.compute_breadcrumbs.return_value = self._expected_breadcrumbs
 
+    def call_get_breadcrumbs(self, system_id: str):
+        """Calls the SystemRepo `get_breadcrumbs` method"""
 
-def test_update_duplicate_name_within_parent(test_helpers, database_mock, system_repository):
-    """
-    Test updating a System with a duplicate name within the parent System
+        self._obtained_system_id = system_id
+        self._obtained_breadcrumbs = self.system_repository.get_breadcrumbs(system_id, session=self.mock_session)
```

Comment:
I can't see this one any more, but I think it may have been fixed by resolving some other type errors. Could you recheck this once I have committed please.